### ISSUE TITLE
Shells warming up

### ISSUE THREAD

**Issue by tobiashuber86** [pyzo/pyzo#644]
I have an issue since a few days. I can't run a shell anymore. It says "Warming up..."
I run on Win 10, 64 bit, Pyzo v4.8.0, Python 3.7.1, Qt5.11.1
I found online somewhere that this could be a firewall issue, but even if I turn the firewall off, it still persists.
It came a few days ago. Could be that there was a Windows update.

----

Summary so far on this issue (by AK):

*update: this has (very likely) been fixed - a new release (v4.9.0) is now available*

* This problem has been reported here, in #643, and [in the mailing list](https://groups.google.com/forum/#!topic/pyzo/Q-DW8xfKwcM). All on Windows 10.
* For @damianquijano it was enough to increase `max_tries` and drop `+256`. For others it was not - it even resulted in a fatal error.
* Some suggest the problem started appearing after a Windows update.
* The problem is consistent for different versions of (binary) Pyzo.
* For some tweaks, the "warming up" is gone, but the shell does not start. This could mean that the issues was fixed for the IDE-connection, but the kernel-connection is having the problem too.

**Comment by almarklein** [comment]
This looks a lot like #634 indeed. I thought that it was a rare glitch, maybe it's not. 

Could you try and change this line https://github.com/pyzo/pyzo/blob/master/pyzo/core/kernelbroker.py#L438:
```py
# c = self._context.bind(address+':PYZO+256', max_tries=32) replaced by
c = self._context.bind(address+':PYZO', max_tries=32)
```

If you're using the binary distribution Pyzo, you can edit the code in the directory `dir_where_pyzo_is_installed/source/pyzo/core/kernelbroker.py`.

**Comment by tobiashuber86** [comment]
This results in: 
Fatal error detected
Failed to execute script __main__

**Comment by almarklein** [comment]
That's a pretty hard failure to start Pyzo. Could it be that you made a typo there, rendering the syntax invalid?

**Comment by tobiashuber86** [comment]
No typo.
I completely swiped my anaconda and pyzo and reinstalled everything.
Still only warming up, but the fatal error is gone (but it is still warming up, even with the changed line 438 in kernelbroker.py)

**Comment by almarklein** [comment]
What about 
```py
c = self._context.bind(address+':PYZO', max_tries=64)
```
?

**Comment by almarklein** [comment]
> warming up is gone, and it looks like there is a shell, but it doesn't start.

This is actually interesting. Could you please try something else:
Change [line 438](https://github.com/pyzo/pyzo/blob/master/pyzo/core/kernelbroker.py#L438) to
```py
c = self._context.bind(address+':pyzo:256', max_tries=32)
```
and [line 380](https://github.com/pyzo/pyzo/blob/master/pyzo/core/kernelbroker.py#L380) to
```py
self._kernelCon = self._context.bind('localhost:pyzo', max_tries=256, name='kernel')
```
(in both cases just make "PYZO" lowercase). My suspicion/hope is that we previously fixed the part on line 438, but that line 380 was still affected, removing the "warming up", but still have a failing kernel. Fingers crossed.

BTW, I put a summary of what I understand about this issue in the top post. Both Windows machines that I have direct access too are unaffected, unfortunately.

**Comment by tobiashuber86** [comment]
yes. fixed the problem (on v 4.8.0)
Thank you so much for your help!

**Comment by almarklein** [comment]
>  fixed the problem (on v 4.8.0)

That's good to hear! Will await confirmation from another user (contact via email), and if this seems like a fix, I'll apply a patch and do a release asap.

**Comment by almarklein** [comment]
Applied fix in #646. Will do a release today. Leaving this open for the time being, so people searching for this issue will know the status.

**Comment by almarklein** [comment]
v4.9.0 is released
